File: pdfestrian/src/main/scala/org/datakind/ci/pdfestrian/api/Metadata.scala
package org.datakind.ci.pdfestrian.api

/**
  * Metadata is returned predictions from API
  * @param record Identifier for record (usually recordId)
  * @param metaData The labelname of the metadata returned  (e.x. biome)
  * @param value The label of the metadata returned  (e.x. forest)
  * @param sentence Context from where the ML model predicted the value
  * @param sentenceLocation The location in the fulltext sentence can be found
  * @param confidence The probability under the model for the returned label
  */
case class Metadata(record : String,
                    metaData : String,
                    value : String,
                    sentence : String,
                    sentenceLocation : Int,
                    confidence : Double)

File: colandr/lib/extractors/metadata.py
from dataclasses import dataclass


@dataclass
class Metadata:
    """
    Metadata model.

    Attributes:
        record (int): Identifier for record (usually recordId).
        metadata (str): The label name of the metadata returned  (e.x. biome).
        value (str): The label of the metadata returned  (e.x. forest).
        sentence (str): Context from where the ML model predicted the value.
        sentence_location (int): The location in the fulltext sentence can be found.
        confidence (float): The probability under the model for the returned label.
        confidence_level (int): The confidence level.
    """
    record: int
    metadata: str
    value: str
    sentence: str
    sentence_location: int
    confidence: float
    confidence_level: int = -1
